expression/builtin_string.go

```diff
@@ -1979,25 +1979,38 @@ type toBase64FunctionClass struct {
 }
 
 func (c *toBase64FunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinToBase64Sig{newBaseBuiltinFunc(args, ctx)}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpString)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	bf.tp.Flen = base64NeededEncodedLength(bf.args[0].GetType().Flen)
+	sig := &builtinToBase64Sig{baseStringBuiltinFunc{bf}}
 	return sig.setSelf(sig), errors.Trace(c.verifyArgs(args))
 }
 
 type builtinToBase64Sig struct {
-	baseBuiltinFunc
+	baseStringBuiltinFunc
+}
+
+// base64NeededEncodedLength return the base64 encoded string length
+func base64NeededEncodedLength(n int) int {
```

Comment:
@XuHuaiyu  But we should use 'base64_needed_encoded_length` cause to_base64 is a base64 encode function.
see https://github.com/mysql/mysql-server/blob/5.7/sql/item_strfunc.cc#L676